inappmessaging/src/main/java/com/rakuten/tech/mobile/inappmessaging/runtime/utils/MessageEventReconciliationUtil.kt

```diff
@@ -108,13 +108,14 @@ internal interface MessageEventReconciliationUtil {
                 // Create an empty list which will contain reconciled events. They will be removed
                 // from the relevantEventsCopy iff Trigger is reconciled.
                 val eventsToBeRemoved = ArrayList<Event>() // NOPMD
-
+                var isPersistentType = false
                 // Reconcile each trigger with all relevant events.
                 for (event in relevantEventsCopy) {
+                    isPersistentType = event.isPersistentType()
```

Comment:
So what if you had a case where `relevantEventsCopy` looked like this: 
```
  Event 1 = {isReconciled = true, isPersistentType = true},
  Event 2 = {isReconciled = false, isPersistentType = false}
```

In that case, the loop would hit the first item which has `isPersistentType=true`, which means it should `continue` at line 135. However, in this case it wouldn't continue because `isPersistentType` was later set to false by the second item in the array. Is that correct behavior?

Reply:
In this part of the process, `relevantEventsCopy` contains list of events with the same type.
Therefore all the events in the list will have the same persistent type.
Also, in the current implementation, persistent type events are only stored locally once. Therefore the list will only contain 1 event.

Reply:
Oh, in that case couldn't you simplify/clarify this by keeping the logic in one place (at line 118) and then you don't have to store the `isPersistentType` variable? Because it looks like the condition `(isPersistentType && numTriggersSatisfied > 0)` will always be true when `isPersistentType` is true. You can use [continue/break labels](https://kotlinlang.org/docs/reference/returns.html):

```
if (event.isPersistentType()) {
  continue@triggers
} else {
  eventsToBeRemoved.add(event)
}
```

Reply:
you're right, I forgot kotlin feature.
thanks, updated code.